### PR TITLE
[MatrixCamp] Add built-in function lengthUTF8

### PR DESCRIPTION
**Pull request by coderzc**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #2116 

**What this PR does / why we need it:**

Add string lengthUTF8(str) built-in function

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by broccoliSpicy — APPROVED** [review]
LGTM

**Comment by domingozhang** [comment]
BVT passed

**Review by yingfeng — CHANGES_REQUESTED** [review]
This implementation is too naive and not efficient. From now on, any string relevant built-in functions should be able to provide algorithms to guarantee the necessary performance. 
Some references:
[ref1](https://github.com/StarRocks/starrocks/blob/main/be/src/util/utf8.h)
[ref2](https://github.com/satanson/cpp_etudes/blob/master/include/string_functions.hh)
you may not provide the simd acceleration, but even the naive version should not just be a simple call on the golang system functions.

**Comment by coderzc** [comment]
> efficient

Thank you for your advice. I have provided the optimized algorithm implementation, please review it again.

**Comment by coderzc** [comment]
> This implementation is too naive and not efficient. From now on, any string relevant built-in functions should be able to provide algorithms to guarantee the necessary performance. Some references: [ref1](https://github.com/StarRocks/starrocks/blob/main/be/src/util/utf8.h) [ref2](https://github.com/satanson/cpp_etudes/blob/master/include/string_functions.hh) [ref3](https://github.com/ClickHouse/ClickHouse/blob/master/src/Common/UTF8Helpers.h) you may not provide the simd acceleration, but even the naive version should not just be a simple call on the golang system functions.

Thank you for your advice. I have provided the optimized algorithm implementation, please review it again.

File: pkg/vectorize/lengthutf8/length_utf8.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		0xb, 0x6, 0x6, 0x6, 0x5, 0x8, 0x8, 0x8, 0x8, 0x8, 0x8, 0x8, 0x8, 0x8, 0x8, 0x8, // f0..ff
+		0x0, 0x1, 0x2, 0x3, 0x5, 0x8, 0x7, 0x1, 0x1, 0x1, 0x4, 0x6, 0x1, 0x1, 0x1, 0x1, // s0..s0
+		1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 1, // s1..s2
+		1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, // s3..s4
+		1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 3, 1, 1, 1, 1, 1, 1, // s5..s6
+		1, 3, 1, 1, 1, 1, 1, 3, 1, 3, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, // s7..s8
+	}
+}
+
+// this countUTF8CodePoints implement bases on https://bjoern.hoehrmann.de/utf-8/decoder/dfa/
```
Comment (broccoliSpicy) [inline pkg/vectorize/lengthutf8/length_utf8.go:49]: // this countUTF8CodePoints implementation, should be a noun

**Review by broccoliSpicy — APPROVED** [review]
cool

**Review by yingfeng — APPROVED** [review]

**Review by broccoliSpicy — APPROVED** [review]
neat

**Review by aunjgr — APPROVED** [review]
The code looks great to me.

**Pull request merged**